src/main/java/app/gpx_animator/ui/cli/CommandLineConfigurationFactory.java

```diff
@@ -348,6 +361,41 @@ private void normalizeTrimGpxEnd() {
         }
     }
 
+    private static String checkVersion(final ResourceBundle resourceBundle) {
+        final var currentVersion = new DefaultArtifactVersion(Constants.VERSION.replace("-SNAPSHOT", ""));
+
+        try {
+            final var dbf = DocumentBuilderFactory.newInstance();
+            final var db = dbf.newDocumentBuilder();
+            final var doc = db.parse(new URL(Constants.UPDATES_URL).openStream());
+            doc.getDocumentElement().normalize();
+
+            final var entries = doc.getElementsByTagName("entry");
```

Comment:
Sorry for my late answer, I am on easter vacation.

This code can be much easier. Every supported OS has always the same version number. So you do not need to check for the OS we are actually running on. It is a nice, working code, but makes this process more complicated than needed.

I would suggest just to load the XML and use a regex to extract the version number from the first occurrence: `newVersion="1.6.1"`